extensions/amp-access-scroll/0.1/scroll-impl.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {CSS} from '../../../build/amp-access-scroll-0.1.css';
+import {AccessClientAdapter} from '../../amp-access/0.1/amp-access-client';
+import {Services} from '../../../src/services';
+import {installStylesForDoc} from '../../../src/style-installer';
+
+const TAG = 'amp-access-scroll-elt';
+
+/** @const {!JsonObject} */
+const CONFIG = /** @type {!JsonObject} */ ({
+  'authorization': 'https://connect.scroll.com/amp/access?' +
+                   'rid=READER_ID&o=SOURCE_URL&' +
+                   'cid=CLIENT_ID(cid-fallback-cookie)',
+  'pingback': 'https://connect.scroll.com/amp/pingback?' +
+              'rid=READER_ID&o=SOURCE_URL&cid=CLIENT_ID(cid-fallback-cookie)&' +
+              'd=AUTHDATA(scroll)&v=AUTHDATA(visitId)',
+  'namespace': 'scroll',
+});
+
+/**
+ * amp-access vendor that authenticates against the scroll.com service.
+ * If the user is authenticated, also adds a fixed position iframe
+ * to the page.
+ *
+ * A little gross, but avoid some duplicate code by inheriting
+ * from ClientAdapter.
+ * @implements {../../amp-access/0.1/access-vendor.AccessVendor}
+ */
+export class ScrollAccessVendor extends AccessClientAdapter {
+  constructor(ampdoc, accessService) {
+    super(ampdoc, CONFIG, {
+      buildUrl: accessService.buildUrl.bind(accessService),
+      collectUrlVars: accessService.collectUrlVars_.bind(accessService),
+    });
+    this.element_ = null;
+  }
+
+  authorize() {
+    return super.authorize()
+        .then(response => {
+          if (response && response.scroll) {
+            this.element_ = new ScrollElement(this.ampdoc);
+            this.ampdoc.getBody().appendChild(this.element_);
+            this.element_.show();
+          }
+        });
+  }
+}
+
+/**
+ * @extends {HTMLElement}
+ */
+class ScrollElement extends HTMLElement {
+  constructor(ampdoc) {
+    super();
+    installStylesForDoc(ampdoc, CSS, () => {}, false, TAG);
+
+    this.ampdoc = ampdoc;
+
+    this.wrapper_ = document.createElement('div');
+    this.wrapper_.classList.add('amp-access-scroll-bar');
+    this.appendChild(this.wrapper_);
+
+
+    this.placeholder_ = document.createElement('div');
+    this.placeholder_.classList.add('amp-access-scroll-placeholder');
+    this.wrapper_.appendChild(this.placeholder_);
+
+    const img = document.createElement('amp-img');
```

Comment:
Also, totally fine to use just normal `<img>`, especially for a logo. This kind of logo might just be better to do via CSS background, but up to you.

Reply:
just changed to img for now